u8timeseries/backtesting/backtesting.py

```diff
@@ -259,3 +260,193 @@ def plot_residuals_analysis(residuals: TimeSeries, num_bins: int = 20):
+    function) is then chosen for that type of model. This way, none of the models have 'looked at' the test set
+    before the final evaluation.
+
+    The performance of every model type is then plotted against the series of actual observations.
+    Additionally, the `metric` values and the runtimes of every model type are plotted in bar charts.
+
+    Parameters
+    ----------
+    train_series
+        A TimeSeries instance used for training during model tuning and evaluation.
+    val_series
+        A TimeSeries instance used for validation during model tuning and for training during evaluation.
+    test_series
+        A TimeSeries instance used for validation when evaluating a model.
+    width
+        An integer indicating the number of plots that are displayed in one row.
+    metric:
+        A function that takes two TimeSeries instances as inputs and returns a float error value.
+    verbose:
+        Whether to print progress.
+    """
+
+    raise_if_not(width > 1, "Please choose an integer 'width' value larger than 1", logger)
+
+    # list of tuples containing model classes and hyperparameter selection, if required
+    model_parameter_tuples = [
+        (NaiveSeasonal, {
+            'K': list(range(1, 31))
+        }),
+        (FFT, {
+            'nr_freqs_to_keep': [5, 10, 25, 50, 100, 150, 200],
+            'trend': [None, 'poly', 'exp']
+        }),
+        (Theta, {
+            'theta': [0, 1] + list(range(3, 50))
+        }),
+        (Prophet, {}),
+        (AutoARIMA, {})
+    ]
+
+    # set up plot grid
+    height = math.ceil(len(model_parameter_tuples) / width) + 1
+    fig = plt.figure(constrained_layout=True, figsize=(width * 4, height * 4))
+    gs = fig.add_gridspec(height, width)
+
+    # arrays to store data for bar charts
+    mape_vals = []
+    times = []
+    index = []
+
+    # iterate through model type selection
+    iterator = _build_tqdm_iterator(model_parameter_tuples, verbose)
+    for i, (model_class, params) in enumerate(iterator):
+        
+        # if necessary, tune hyperparameters using train_series and val_series
+        if (len(params.keys()) > 0):
+            model = backtest_gridsearch(model_class, params, train_series, val_series, metric=metric)
+        else:
+            model = model_class()
+
+        # fit on train and val series, predict test series
+        train_and_val_series = train_series.append(val_series)
+        start_time = time.time()
+        model.fit(train_and_val_series)
+        end_time = time.time()
+        predictions = model.predict(len(test_series))
+
+        # plot predictions against observations
+        y = int(i / width)
+        x = i - y * width
+        ax = fig.add_subplot(gs[y, x])
+        mape_val = metrics.mape(predictions, test_series)
+        ax.set_title(str(model) + "\n" + metric.__name__ + ": " + str(round(mape_val, 2)))
+        train_and_val_series[-len(test_series):].plot(label='train and val', ax=ax)
```

Comment:
Is it correct to take this indexing `[-len(test_series):]` ?

Reply:
I did this so that half of the graph (in terms of the x axis) would be reserved for the prediction, so that it is clearly visible. Or do you mean to ask whether this form of indexing even works on TimeSeries objects? If that's the case then yes, it seems to be working perfectly.

Reply:
No I meant to ask if it was really your intent to plot only the `len(test_series)` last points. I think it might be weird, especially for tiny test series. I would find it more intuitive to plot the whole thing here (and let the users zoom in if needed).

Reply:
done